core/shared/src/main/scala/org/http4s/Credentials.scala

```diff
@@ -102,21 +102,26 @@ object BasicCredentials {
   private def decode(bytes: Array[Byte], charset: JavaCharset): String =
     new String(bytes, charset)
 
-  def apply(token: String): BasicCredentials = {
-    val bytes = Base64.getDecoder.decode(token)
-    val (userPass, charset) = decode(bytes, utf8CharsetDecoder)
-      .fold(_ => (decode(bytes, fallbackCharset), fallbackCharset), up => (up, utf8Charset))
-    userPass.indexOf(':') match {
-      case -1 => apply(userPass, "", charset)
-      case ix => apply(userPass.substring(0, ix), userPass.substring(ix + 1), charset)
+  @deprecated("Use fromString instead", "0.23.24")
+  def apply(token: String): BasicCredentials =
+    fromString(token).get
```

Comment:
If I'm not wrong, previously there was an `IllegalArgumentException`, and now it will be `NoSuchElementException`. Although the difference is subtle, shouldn't we preserve the behaviour?

Reply:
Good point!

Reply:
Fixed this one.